zaius/s3/utils.py

```diff
@@ -0,0 +1,63 @@
+import boto3
+import os
+from functools import partial
+from multiprocessing import Pool, cpu_count
+
+import zaius.auth as auth
+
+class S3Client():
+
+    def __init__(self, auth_struct=None):
+        self.auth = auth_struct if auth_struct is not None else auth.default()
+        self.client = init_s3_client(self.auth)
+
+    def download_from_s3(self, bucket, local_path, key):
+        """
+        downloads a file from s3
+        """
+        download_from_s3(bucket, local_path, key)
```

Comment:
it feels that you can simply move the body of the corresponding functions here, so you don't need to refer to the functions in the class (it seems you use the class anyway)

Reply:
Yer so I have done it this way because you can't parallelise functions that are defined inside of a class. It's a bit messy I know but it also means you can either import a single function or use the auth handling of the class